### ISSUE TITLE
Index out-of-range access to word_segment_list  

### ISSUE THREAD

**Issue by syoyo** [m-bain/whisperX#22]
@m-bain Thank you for whisperX!

In some audio,  I found `IndexError: list index out of range` error happens at aligning stage.

```
Traceback (most recent call last):                                                                            
  File "/home/syoyo/miniconda3/envs/whisperx/bin/whisperx", line 8, in <module>                               
    sys.exit(cli())                                                                                           
  File "/home/syoyo/miniconda3/envs/whisperx/lib/python3.8/site-packages/whisperx/transcribe.py", line 505, in
 cli                                                                                                          
    result_aligned = align(result["segments"], align_model, align_metadata, audio_path, device,              
  File "/home/syoyo/miniconda3/envs/whisperx/lib/python3.8/site-packages/whisperx/transcribe.py", line 374, in
 align                                                                                                        
    word_segments_list[-1]['text'] += ' ' + curr_word                                                        
IndexError: list index out of range 
```

## How to reproduce

https://commonvoice.mozilla.org/en/datasets

Download Japanese -> Common Voice Copus 12.0

![image](https://user-images.githubusercontent.com/18676/211053790-a86b8eef-902e-4e91-9702-daa05f786377.png)

```
$ whisperx --model large --language ja cv-corpus-12.0-2022-12-07/ja/clips/common_voice_ja_35797612.mp3
```

## Investigation

https://github.com/m-bain/whisperX/blob/2aa074e0e63d4b2da0f40dc30dfa62f04c9dc6c3/whisperx/transcribe.py#L382

Here is the dump of `t_local` and `t_words` before `for x in range(len(t_local)):`

```
t_local = [None, None, (0.842688, 1.083456), (1.424544, 1.5048), (1.524864, 1.544928), (1.665312, 1.7455679999999998), (1.7656319999999999, 1.785696), (1.865952, 2.026464), (2.046528, 2.186976), (2.20704, 2.2271039999999998), (2.247168, 2.327424), (2.347488, 2.367552), (2.427744, 2.508), (2.548128, 2.568192), (2.688576, 2.969472), None, (3.089856, 3.10992), (3.129984, 3.2102399999999998), (3.31056, 3.8322239999999996), (3.852288, 3.9325439999999996), (4.052928, 4.072992), (4.133184, 4.2535680000000005), (4.333824, 4.434144), (4.494336, 4.715039999999999), (4.735104, 4.755168), (4.775232, 5.196576), (5.31696, 5.457407999999999), (5.597856, 5.678112), (5.738303999999999, 5.81856), (5.838624, 5.898815999999999), (5.9790719999999995, 6.0793919999999995), (6.159648, 6.199776), (6.21984, 6.280032), (6.360288, 6.460608000000001), (6.480671999999999, 6.5207999999999995), (6.641183999999999, 6.761568), (6.781632, 6.801696), None]
t_words = ['童', '貞', '助', 'か', 'ら', 'な', 'い', 'と', '思', 'っ', 'て', 'い', 'る', 'か', 'ら', '、', 'い', 'る', 'と', 'う', 'っ', 'と', 'さ', 'り', 'っ', 'と', '音', 'が', 'し', 'て', '目', 'か', 'ら', '火', 'が', '出', 'た', '。']
```

Probably we need to consider a situation `t_local` may contain multiple `None` s from the start?

**Comment by m-bain** [comment]
Thank you for reporting the error, I have fixed the logic in the recent commit [857bcca](https://github.com/m-bain/whisperX/commit/857bcca238b2dad1439f53d15d3566b5371c5a54), seems to work now with the file you referenced

**Comment by syoyo** [comment]
@m-bain Thanks! Confirmed the recent commit fixes this issue. Closing!
